test/e2e/vmagent_test.go

```diff
@@ -71,9 +73,134 @@ func TestVMAgent(t *testing.T) {
 	t.Run("testExternalNodeWithANP", func(t *testing.T) { testExternalNodeWithANP(t, data, vmList) })
+	t.Run("testExternalNodeSupportBundleCollection", func(t *testing.T) { testExternalNodeSupportBundleCollection(t, data, vmList) })
+}
+
+func (data *TestData) waitForDeploymentRealized(t *testing.T, namespace string, name string, timeout time.Duration) error {
+	t.Logf("Waiting for Deployment '%s/%s' to be realized", namespace, name)
+	if err := wait.Poll(100*time.Millisecond, timeout, func() (bool, error) {
+		dp, err := data.clientset.AppsV1().Deployments(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return dp.Status.ObservedGeneration == dp.Generation && dp.Status.ReadyReplicas == *dp.Spec.Replicas, nil
+	}); err != nil {
+		return fmt.Errorf("error when waiting for Deployment '%s/%s' to be realized: %v", namespace, name, err)
+	}
+	return nil
+}
+
+func (data *TestData) waitForSupportBundleCollectionRealized(t *testing.T, name string, timeout time.Duration) error {
+	t.Logf("Waiting for SupportBundleCollection '%s' to be realized", name)
+	var sbc *crdv1alpha1.SupportBundleCollection
+	if err := wait.Poll(100*time.Millisecond, timeout, func() (bool, error) {
+		sbc, conditionErr := data.crdClient.CrdV1alpha1().SupportBundleCollections().Get(context.TODO(), name, metav1.GetOptions{})
+		if conditionErr != nil {
+			return false, conditionErr
+		}
+		return sbc.Status.DesiredNodes == sbc.Status.CollectedNodes, nil
+	}); err != nil {
+		if sbc != nil {
+			t.Logf("The conditions of supportbundlecollection for the vms are %v", sbc.Status.Conditions)
+		}
+		return fmt.Errorf("error when waiting for SupportBundleCollection '%s' to be realized: %v", name, err)
+	}
+	return nil
+}
+
+func testExternalNodeSupportBundleCollection(t *testing.T, data *TestData, vmList []vmInfo) {
+	sftpServiceYAML := "sftp-deployment.yml"
+	secretUserName := "foo"
+	secretPassword := "pass"
+	uploadFolder := "upload"
+	uploadPath := path.Join("/home", secretUserName, uploadFolder)
+	secretName := "support-bundle-secret"
+	vmNames := make([]string, 0, len(vmList))
+	for _, vm := range vmList {
+		vmNames = append(vmNames, vm.nodeName)
+	}
+	code, _, stderr, err := data.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl apply -f %s -n %s", sftpServiceYAML, data.testNamespace))
+	if code != 0 {
+		t.Errorf("Error When applying %s: %v", sftpServiceYAML, stderr)
+	}
+	if err != nil {
+		t.Errorf("Error When applying %s: %v", sftpServiceYAML, err)
+	}
+	failOnError(data.waitForDeploymentRealized(t, data.testNamespace, "sftp", 30*time.Second), t)
+	sec := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: secretName,
+		},
+		Data: map[string][]byte{
+			"username": []byte(secretUserName),
+			"password": []byte(secretPassword),
+		},
+	}
+	_, err = data.clientset.CoreV1().Secrets(namespace).Create(context.TODO(), sec, metav1.CreateOptions{})
```

Comment:
ditto

Reply:
https://github.com/antrea-io/antrea/blob/main/ci/jenkins/test-vm.sh#L115 will clean resources in vm-ns namespace

Reply:
That's not symmetric and just adds unnecessary dependency on test script. Isn't better to delete what a test creates in the end of the test?

Reply:
updated.